language/stdlib/modules/0L/Wallet.move

```diff
@@ -0,0 +1,411 @@
+    use 0x1::CoreAddresses;
+    use 0x1::Vector;
+    use 0x1::Signer;
+    use 0x1::Errors;
+    use 0x1::LibraConfig;
+    use 0x1::Option::{Self,Option};
+    use 0x1::LibraSystem;
+    use 0x1::NodeWeight;
+    use 0x1::Debug::print;
+
+    const ERR_PREFIX: u64 = 023;
+
+    //////// COMMUNITY WALLETS ////////
+
+    resource struct CommunityWallets {
+        list: vector<address>
+    }
+
+        // Timed transfer submission
+    resource struct CommunityTransfers {
+      proposed: vector<TimedTransfer>,
+      approved: vector<TimedTransfer>,
+      rejected: vector<TimedTransfer>,
+      max_uid: u64,
+
+    }
+    struct TimedTransfer {
+      uid: u64,
+      expire_epoch: u64,
+      payer: address,
+      payee: address,
+      value: u64,
+      description: vector<u8>,
+      veto: Veto,
+    }
+
+    struct Veto {
+      list: vector<address>,
+      count: u64,
+      threshold: u64,
+    }
+
+    resource struct CommunityFreeze {
+        is_frozen: bool,
+        consecutive_rejections: u64,
+        unfreeze_votes: vector<address>,
+    }
+
+    // Utility used at genesis (and on upgrade) to initialize the system state.
+    public fun init(vm: &signer) {
+        CoreAddresses::assert_libra_root(vm);
+        
+        if ((!exists<CommunityTransfers>(0x0))) {
+          move_to<CommunityTransfers>(vm, CommunityTransfers{
+            proposed: Vector::empty<TimedTransfer>(),
+            approved: Vector::empty<TimedTransfer>(),
+            rejected: Vector::empty<TimedTransfer>(),
+            max_uid: 0,
+          })
+        }; 
+
+      if (!exists<CommunityWallets>(0x0)) {
+        move_to<CommunityWallets>(vm, CommunityWallets {
+          list: Vector::empty<address>()
+        });  
+      }
+    }
+
+    public fun set_comm(sig: &signer) acquires CommunityWallets {
+      if (exists<CommunityWallets>(0x0)) {
+        let addr = Signer::address_of(sig);
+        let list = get_comm_list();
+        if (!Vector::contains<address>(&list, &addr)) {
+            let s = borrow_global_mut<CommunityWallets>(0x0);
+            Vector::push_back(&mut s.list, addr);
+        };
+
+        move_to<CommunityFreeze>(sig, CommunityFreeze {
+          is_frozen: false,
+          consecutive_rejections: 0,
+          unfreeze_votes: Vector::empty<address>()
+        })
+      }
+    }
+
+
+    // Utility for vm to set an address as a Community Wallet
+    public fun vm_set_comm(vm: &signer, addr: address) acquires CommunityWallets {
+      CoreAddresses::assert_libra_root(vm);
+      if (exists<CommunityWallets>(0x0)) {
+        let list = get_comm_list();
+        if (!Vector::contains<address>(&list, &addr)) {
+        
+          let s = borrow_global_mut<CommunityWallets>(0x0);
+          Vector::push_back(&mut s.list, addr);
+        }
+      }
+    }
+
+    // Utility for vm to remove the CommunityWallet tag from an address
+    public fun vm_remove_comm(vm: &signer, addr: address) acquires CommunityWallets {
```

Comment:
Will this ever be used? I can't think of such a situation unless the whole idea of community wallets is removed.

Reply:
This will need to exist if there is a vote to remove. It's not implemented anywhere, just future proofing.